docs/zh/design/lock/lock-api-design.md

```diff
@@ -0,0 +1,262 @@
+# 0. 太长，不看
+添加TryLock和Unlock API. 
+
+续租API有争议，第一版不加入续租API
+
+# 1. 调研
+| **系统** | **能否实现分布式锁** | **阻塞锁(基于watch)** | **可用性** | **写操作线性一致** | **sequencer([chubby论文里提出的feature]([https://static.googleusercontent.com/media/research.google.com/zh-TW//archive/chubby-osdi06.pdf](https://static.googleusercontent.com/media/research.google.com/zh-TW//archive/chubby-osdi06.pdf)))** | **续租** |
+| --- | --- | --- | --- | --- | --- | --- |
+| 单机redis | yes | x | 单点失效时，锁服务不可用 | yes | yes(need poc) | yes |
+| redis集群 | yes | x | yes | no. 故障转移可能导致丢锁 | yes(need poc) | yes |
+| redis Redlock | yes |  |  |  |  |  |
+| nacos | × |  |  |  |  |  |
+| consul | yes |  |  |  |  |  |
+| eureka | × |  |  |  |  |  |
+| zookeeper | yes | yes | yes 有fo能力,[200 ms内完成选举](https://pdos.csail.mit.edu/6.824/papers/zookeeper.pdf) | yes | yes 使用zxid作为sequencer | yes |
```

Comment:
it's weird to put nacos and eureka here as they are simply service registries and has nothing to do with distributed lock

Reply:
fixed. nacos and eureka have been removed